sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -530,6 +523,16 @@ class ResolveSessionCatalog(
         replace,
         viewType)
 
+    case ShowViewsStatement(SessionCatalogAndNamespace(_, ns), pattern) =>
```

Comment:
what's the error message if SHOW VIEWS refers to a custom catalog?

Reply:
Actually, I'm not quite sure about the question. The `SHOW VIEWS` will resolve to V1 SessionCatalog which supports `listViews` I think. Do you mean a custom catalog implements `TableCatalog`(does not support view related commands) will got error? I'll keep learning the `CatalogPlugIn` related code, any suggestions would be helpful to me, thanks!

Reply:
Because here we only catch `SessionCatalogAndNamespace`, so if SHOW VIEWS refers to a custom catalog, we may hit Scala match not found exception, which is a bad error message to users.

Reply:
I see, thanks so much for your clarification, @cloud-fan ! `SessionCatalogAndNamespace` is a support function to make sure the resolved catalog is `SessionCatalog`, which supports several V1 commands. So I think maybe we should just throw `AnalysisException` in `SessionCatalogAndNamespace`? WDYT, updated in d14b4fa698873cf0eef4f1ca50b9a501c5cd868e.

Reply:
Thanks so much for the clarification, @cloud-fan ! I did some try and still working on this, will update soon.

Reply:
Updated and added a negative test case in `DataSourceV2SQLSuite`